scraper/src/merge_datasets.js

```diff
@@ -2,28 +2,162 @@ import fs from "fs";
-const fetchGermanNames = async (plants) => {
-  return Promise.all(
-    plants.map(async (plant) => {
-      if (plant["common_name_de"]) {
-        return plant;
+axiosRetry(axios, {
+  retries: 5, // number of retries
+  retryDelay: (retryCount) => {
+    console.log(`retry attempt: ${retryCount}`);
+    return retryCount * 3000; // time interval between retries
+  },
+});
+
+/**
+ * Fetches the German name of the plant from the Wikidata API.
+ * Sets the 'common_name_de' property of every plant in the array.
+ *
+ * @param {string[]} germanNames - An array with German plant names.
+ * @param {string} unique_name - A plant name to filter if it's in the German names.
+ */
+const filterGermanNames = (germanNames, unique_name) => {
+  let unique_name_filter = unique_name.toLowerCase();
+  unique_name_filter = unique_name_filter.replace(/(?<!\S)[x×]/, " ");
+  unique_name_filter = unique_name_filter.replace(/\s+/g, " ");
+
+  const cleanedGermanNames = [];
+
+  for (const singleGermanName of germanNames) {
+    let germanName = singleGermanName.toLowerCase();
+
+    germanName = germanName.replace(/ \(.*\)/, "");
+    //remove special characters
+    germanName = germanName.replace(/['"`-]/, "");
+    //remove hybrid x or × symbol
+    germanName = germanName.replace(/(?<!\S)[x×]/, " ");
+    unique_name_filter = unique_name_filter.replace(/\s+/g, " ");
+
+    germanName = germanName.trim();
+    if (germanName.includes(unique_name_filter)) {
+      continue;
+    }
+    if (germanName.length !== 0 && germanName !== "true") {
+      cleanedGermanNames.push(capitalizeWords(germanName));
+    }
+  }
+  //remove duplicates
+  const uniqueNameSet = new Set(cleanedGermanNames);
+
+  return Array.from(uniqueNameSet);
+};
+
+/**
+ * Fetches the German name of the plant from the Wikidata API.
+ * Sets the 'common_name_de' property of every plant in the array.
+ *
+ * @param {Array} plants - An array containing plant objects.
+ */
+const fetchGermanNamesForPlantsConcurrent = async (plants) => {
+  const MAX_CONCURRENT_REQUESTS = 25;
+  let GermanNamesFound = 0;
+
+  console.log("[INFO] Start fetching German common names!");
+
+  const processPlants = async (plants) => {
+    for (const plant of plants) {
+      const unique_name = plant["unique_name"];
+      const germanNames = [];
+
+      if (plant.common_name_de && plant.common_name_de !== "true") {
+        const existingGermanNames = plant.common_name_de.split(",");
+        existingGermanNames.forEach((existingGermanName) => {
+          germanNames.push(existingGermanName);
+        });
       }
-      const germanName = await fetchGermanName(plant["unique_name"]);
-      plant["common_name_de"] = germanName;
-      return plant;
-    })
-  );
+
+      plant.common_name_de = null;
+
+      await axios
+        .get(
+          `https://www.wikidata.org/w/api.php?action=wbgetentities&sites=enwiki&titles=${unique_name}&normalize=&languages=de&format=json`
+        )
+        .then((response) => {
```

Comment:
The callback in .then could also be factored out into its own function for improved readability.